src/components/pages/Settings/Expansions/ContentCustomization/index.tsx

```diff
@@ -35,19 +36,93 @@ const mapStateToProps = (state: RootState, { expansionId }: OwnProps) => {
       state,
       { expansionId }
     ),
+    allNemesesSelected: selectors.Settings.Expansions.Nemeses.getAllNemesesOfExpansionSelected(
+      state,
+      { expansionId }
+    ),
+    allMagesSelected: selectors.Settings.Expansions.Mages.getAllMagesOfExpansionSelected(
+      state,
+      { expansionId }
+    ),
+    allSupplyCardsSelected: selectors.Settings.Expansions.Cards.getAllCardsOfExpansionSelected(
+      state,
+      { expansionId }
+    ),
+    allTreasuresSelected: selectors.Settings.Expansions.Treasures.getAllTreasuresOfExpansionSelected(
+      state,
+      { expansionId }
+    ),
+    allBasicNemesisCardsSelected: selectors.Settings.Expansions.BasicNemesisCards.getAllBasicNemesisCardsOfExpansionSelected(
+      state,
+      { expansionId }
+    ),
+    allUpgradedBasicNemesisCardsSelected: selectors.Settings.Expansions.UpgradedBasicNemesisCards.getAllUpgradedBasicNemesisCardsOfExpansionSelected(
+      state,
+      { expansionId }
+    ),
   }
 }
 
 const mapDispatchToProps = {
   // FIXME use available expansions instead
   selectLanguage: actions.Settings.Expansions.Languages.select,
+  handleSelectAllNemeses: actions.Settings.Expansions.Nemeses.main.toggleAll,
+  handleSelectAllMages: actions.Settings.Expansions.Mages.main.toggleAll,
+  handleSelectAllSupplyCards: actions.Settings.Expansions.Cards.main.toggleAll,
+  handleSelectAllTreasures:
+    actions.Settings.Expansions.Treasures.main.toggleAll,
+  handleSelectAllBasicNemesisCards:
+    actions.Settings.Expansions.BasicNemesisCards.main.toggleAll,
+  handleSelectAllUpgradedBasicNemesisCards:
+    actions.Settings.Expansions.UpgradedBasicNemesisCards.main.toggleAll,
 }
 
 type Props = ReturnType<typeof mapStateToProps> &
   typeof mapDispatchToProps &
   OwnProps
 
-const ContentCustomization = ({ expansionId, lang, selectLanguage }: Props) => {
+const ContentCustomization = ({
+  expansionId,
+  lang,
+  allNemesesSelected,
+  allMagesSelected,
+  allSupplyCardsSelected,
+  allTreasuresSelected,
+  allBasicNemesisCardsSelected,
+  allUpgradedBasicNemesisCardsSelected,
+  selectLanguage,
+  handleSelectAllNemeses,
+  handleSelectAllMages,
+  handleSelectAllSupplyCards,
+  handleSelectAllTreasures,
+  handleSelectAllBasicNemesisCards,
+  handleSelectAllUpgradedBasicNemesisCards,
+}: Props) => {
+  const handleCheckboxChange = () => {
+    if (
+      allNemesesSelected &&
+      allMagesSelected &&
+      allSupplyCardsSelected &&
+      allTreasuresSelected &&
+      allBasicNemesisCardsSelected &&
+      allUpgradedBasicNemesisCardsSelected
+    ) {
+      handleSelectAllNemeses(expansionId, 'deselect')
+      handleSelectAllMages(expansionId, 'deselect')
+      handleSelectAllSupplyCards(expansionId, 'deselect')
+      handleSelectAllTreasures(expansionId, 'deselect')
+      handleSelectAllBasicNemesisCards(expansionId, 'deselect')
+      handleSelectAllUpgradedBasicNemesisCards(expansionId, 'deselect')
```

Comment:
we could either combine these into a callback accepting a type, or we could use a top level reducer and `Cmd.list` to dispatch all these actions on `select`/`deselect`

Reply:
Another way would be to add a `handleSelectAllContentOfExpansion` action which each sub-reducer could listen to - this would probably be the best way to do this, as redux is basically designed to behave like this
The actuall handling of this reducer case would look the same as the specific handleSelectAll of the respective reducer.

Reply:
The above method where each reducer handles a single action would be the one I would prefer.
(if you aren't sure how to approach this, I can show you)

Reply:
I implemented an action which is consumed by the sub-reducers. Maybe you could have a look if the implementation is like you mentioned above. Or we have a look together when the PR is ready. :)